### PR TITLE
FIX #11176 xAxis.axisTick.interval appears different between echart4 and echart3.8

### LINKED ISSUE #11176: xAxis.axisTick.interval appears different between echart4 and echart3.8
### Version
4.2.1

### Reproduction link
[https://gallery.echartsjs.com/editor.html?c=xy5OQBi9fa&comment=0](https://gallery.echartsjs.com/editor.html?c=xy5OQBi9fa&comment=0)

### Steps to reproduce
echart4  https://gallery.echartsjs.com/editor.html?c=xy5OQBi9fa&comment=0      
echart3.8 https://gallery.echartsjs.com/editor.html?c=xG60Tajuaz&v=3

### What is expected?
希望echart4 和 echart3.8的效果一样

### What is actually happening?
echart4 的 interval 无法隐藏第二个刻度线

### PR DESCRIPTION
**Pull request by foolzhang**
different display in version 3.6 and 4 with xAxis.axisTick.interval

### CODE REVIEW

**Comment by pissang** [comment]
Thanks @foolzhang ! Can you add add a test in the https://github.com/apache/incubator-echarts/tree/master/test

**Comment by foolzhang** [comment]
how about this time @pissang

**Comment by pissang** [comment]
Scheduled in the version 4.5.0

File: src/coord/Axis.js [hunk 1]
```diff
@@ -315,7 +320,10 @@ function fixOnBandTicksCoords(axis, ticksCoords, tickCategoryInterval, alignWith
                 ticksItem.coord -= shift / ((tickCategoryInterval + 1) * 2);
             }
         });
-        last = {coord: ticksCoords[ticksLen - 1].coord + shift};
+
+        diffSize = tickLen - ticksCoords[ticksLen - 1].tickValue;
```
Comment (pissang) [inline src/coord/Axis.js]: I think this should be 
```js
diffSize = tickLen - crossLen;
```

**Comment by pissang** [comment]
Hi @foolzhang.  I checked the code. Very neat fix! But it breaks the test case with dataZoom in axis-interval2.html.

I've left my change suggestions in the comment. Really nice job!

File: src/coord/Axis.js (hunk 1 above)
Comment (foolzhang) [reply src/coord/Axis.js]: @pissang thanks your time , i am feel sorry about not consider dataZoom , but if like this 
```suggestion
        diffSize = tickLen - ticksCoords[ticksLen - 1].tickValue;
```
in my test demo axis-interval.html 
```suggestion
{
        show: true,   
        interval: function (index) {   
        // [false, true, true, false],
        // [false, true, false, true],
        return testArr[dataIndex][index]   
},
```
axis's lastTick  will last. 
eg:[false, true, true, false]  
image
eg:[false, true, false, true]  
image

i review my submit  , i think init tickLen should like this
`

        var dataExtent = this.scale.getExtent();
        //  obtain all the ticks
        var tickLen = this.scale.getTicks().length + dataExtent[0];
        fixOnBandTicksCoords(
            this, ticksCoords, result.tickCategoryInterval, alignWithLabel, opt.clamp,tickLen
        );
    ... 

`
after that, i test my case  and  axis-interval2.html with dataZoon, it's work now. 
will pull request again, thanks you time again

File: src/coord/Axis.js [hunk 2]
```diff
@@ -198,8 +197,11 @@ Axis.prototype = {
         }, this);
 
         var alignWithLabel = tickModel.get('alignWithLabel');
+        var dataExtent = this.scale.getExtent();
+        var tickLen = this.scale.getTicks().length + dataExtent[0];
+        
         fixOnBandTicksCoords(
-            this, ticksCoords, result.tickCategoryInterval, alignWithLabel, opt.clamp
+            this, ticksCoords, result.tickCategoryInterval, alignWithLabel, opt.clamp,tickLen
         );
```
Comment (foolzhang) [inline src/coord/Axis.js]: init tickLen should consider dataZoom

File: src/coord/Axis.js [hunk 3]
```diff
@@ -197,7 +197,8 @@ Axis.prototype = {
         }, this);
 
         var alignWithLabel = tickModel.get('alignWithLabel');
-        var tickLen = this.scale.getTicks().length;
+        var dataExtent = this.scale.getExtent();
+        var tickLen = this.scale.getTicks().length + dataExtent[0];
```
Comment (foolzhang) [inline src/coord/Axis.js]: major changes

File: src/coord/Axis.js (hunk 1 above)
Comment (pissang) [reply src/coord/Axis.js]: Great! Looks good to me now:)

File: src/coord/Axis.js [hunk 4]
```diff
@@ -198,8 +197,11 @@ Axis.prototype = {
         }, this);
 
         var alignWithLabel = tickModel.get('alignWithLabel');
+        var dataExtent = this.scale.getExtent();
+        var tickLen = this.scale.getTicks().length + dataExtent[0];
+        
         fixOnBandTicksCoords(
-            this, ticksCoords, result.tickCategoryInterval, alignWithLabel, opt.clamp
+            this, ticksCoords, result.tickCategoryInterval, alignWithLabel, opt.clamp,tickLen
```
Comment (Ovilia) [inline src/coord/Axis.js]: Space is required before `tickLen`.

File: src/coord/Axis.js [hunk 5]
```diff
@@ -292,7 +294,7 @@ function fixExtentWithBands(extent, nTick) {
 // splitLine/spliteArea should layout appropriately corresponding
 // to displayed labels. (So we should not use `getBandWidth` in this
 // case).
-function fixOnBandTicksCoords(axis, ticksCoords, tickCategoryInterval, alignWithLabel, clamp) {
+function fixOnBandTicksCoords(axis, ticksCoords, tickCategoryInterval, alignWithLabel, clamp,tickLen) {
```
Comment (Ovilia) [inline src/coord/Axis.js]: Space is required before `tickLen`.

File: src/coord/Axis.js [hunk 6]
```diff
@@ -301,12 +303,16 @@ function fixOnBandTicksCoords(axis, ticksCoords, tickCategoryInterval, alignWith
...
     var last;
+    var diffSize;
     if (ticksLen === 1) {
         ticksCoords[0].coord = axisExtent[0];
         last = ticksCoords[1] = {coord: axisExtent[0]};
     }
     else {
-        var shift = (ticksCoords[1].coord - ticksCoords[0].coord);
+
+        var crossLen = ticksCoords[ticksLen-1].tickValue - ticksCoords[0].tickValue;
```
Comment (Ovilia) [inline src/coord/Axis.js]: Space is required around `-`.

File: test/axis-interval.html [hunk 7]
```diff
@@ -338,6 +337,67 @@ <h1>[ Test time axis interval ]&nbsp;&nbsp;&nbsp; should not overlap when data z
...
+            }]
+        };
+        let ec = echarts.init(document.getElementById('main3'));
+
+        setInterval(function(){
+            let next = ++dataIndex;
+            dataIndex = next > 5 ? 0: next;
+            option.title.text = print(dataIndex);
+            ec.setOption(option);
+        },6000);
```
Comment (Ovilia) [inline test/axis-interval.html]: Space is required before `6000`.

File: test/axis-interval.html [hunk 8]
```diff
@@ -338,6 +337,67 @@ <h1>[ Test time axis interval ]&nbsp;&nbsp;&nbsp; should not overlap when data z
...
+            [false, false, true, true],
+            [true, true, false, false]
+        ];
+
+        function print(index){
+            return `intervalList = [${testArr[index].toString()}] ,currentIndex is ${index}`
+        }
+
+        var option = {
+            title:{
```
Comment (Ovilia) [inline test/axis-interval.html]: Space is required before `:`, similar to other places.

Comment (foolzhang) [reply test/axis-interval.html]: will pay attention next time

Comment (Ovilia) [reply test/axis-interval.html]: @foolzhang Never mind. :) You can run `jshint` next time before you make a pull request.

**Pull request merged**